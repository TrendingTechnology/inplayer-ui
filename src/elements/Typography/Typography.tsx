import React, { ReactChild, CSSProperties } from 'react';
import styled, { DefaultTheme } from 'styled-components';
import { prop } from 'styled-tools';
import { fontWeights, fontSizes } from 'utils';
import { textPrimary, textDanger, textSuccess, textWarning } from 'modifiers';
import { applyStyleModifiers } from 'styled-components-modifiers';
import colors from 'theme/colors';

const typographyModifiers = {
  textPrimary,
  textDanger,
  textSuccess,
  textWarning,
};

export type TypographyVariant = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';

type ParagraphProps = {
  color: string;
  description: boolean;
  modifiers: any;
};

type HeadingProps = {
  color: string;
  variant: keyof {
    h1: string;
    h2: string;
    h3: string;
    h4: string;
    h5: string;
    h6: string;
  };
  theme: DefaultTheme;
  modifiers: any;
};

const Paragraph = styled.p<ParagraphProps>`
  font-size:${({ theme }) => theme.font.sizes.medium};
  color: ${({ color }) => color || `${colors.fontDarkGray}`};
  ${({ description, theme: { palette, font } }) =>
    description &&
    `
     color: ${palette.text.light};
     font-size:${font.sizes.large};
   `}
  ${applyStyleModifiers(typographyModifiers)}
`;

const Heading = styled.h1<HeadingProps>`
  font-family: ${prop('theme.font.primary')};
  font-weight: ${({ theme }) => theme.font.weights.thin}
  color: ${({ color }) => color || colors.blue};
  font-size: ${({ theme, variant }) => theme.font.sizes[variant]};
  ${applyStyleModifiers(typographyModifiers)}
`;

export type TypographyProps = {
  /**
   * Variant type
   */
  variant?: TypographyVariant;
  /**
   * When prop is set, the paragraph element has `palette.primary.fontGray` color
   */
  description?: boolean;
  children: ReactChild;
  /**
   * Text color modifiers
   */
  modifiers?: Array<string>;
  /**
   * Text color
   */
  color?: string;
  /**
   * External CSS class name
   */
  className?: string;
  /**
   * Inline CSS
   */
  style?: CSSProperties;
};

const Typography = ({
  variant = 'h1',
  children,
  description = false,
  className = '',
  modifiers = [],
  color = '',
  style = {},
}: TypographyProps) => {
  if (variant === 'p') {
    return (
      <Paragraph
        className={className}
        description={description}
        modifiers={modifiers}
        color={color}
        style={style}
      >
        {children}
      </Paragraph>
    );
  }
  return (
    <Heading
      as={variant}
      className={className}
      variant={variant}
      modifiers={modifiers}
      color={color}
      style={style}
    >
      {children}
    </Heading>
  );
};

export default Typography;
