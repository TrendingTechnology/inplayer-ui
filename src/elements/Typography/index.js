// @flow
import React, { type Node } from 'react';
import styled, { css } from 'styled-components';
import { prop, ifProp } from 'styled-tools';
import { uiColors, fontWeights, fontSizes } from 'utils';

type TypographyVariant = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';

const Paragraph = styled.p`
  max-width: 720px;
  font-size: ${fontSizes('medium')};
  color: ${uiColors('text.main')};
  ${ifProp(
    'description',
    css`
      color: ${uiColors('text.light')};
      font-size: ${fontSizes('large')};
    `
  )};
`;

const Heading = styled.h1`
  font-family: ${prop('theme.font.primary')};
  font-weight: ${fontWeights('thin')};
  margin: 1.7em 0 1em;
  color: ${uiColors('secondary.main')};
  font-size: ${({ theme, variant }) => theme.font.sizes[variant]};
`;

export type TypographyProps = {
  variant: TypographyVariant,
  description: ?boolean,
  children: Node,
};

const Typography = ({ variant, children, description }: TypographyProps) => {
  if (variant === 'p') {
    return <Paragraph description={description}>{children}</Paragraph>;
  }
  const Component = Heading.withComponent(variant);
  return <Component variant={variant}>{children}</Component>;
};

export default Typography;
