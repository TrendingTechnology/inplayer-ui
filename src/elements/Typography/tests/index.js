import { generateSnapshots } from 'helpers';

import Typography from '../index';

generateSnapshots(Typography);
