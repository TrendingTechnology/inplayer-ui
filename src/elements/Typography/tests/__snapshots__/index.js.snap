// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly 1`] = `
.c0 {
  font-family: Roboto;
  font-weight: 100;
  margin: 1.7em 0 1em;
  color: #286090;
  font-size: 2.5rem;
}

<h1
  className="c0"
  color=""
/>
`;
